openjdk.test.load/src/test.load/net/adoptopenjdk/stf/MixedLoadTest.java

```diff
@@ -63,13 +80,25 @@ public void execute(StfCoreExtension test) throws StfException {
 				.addProjectToClasspath("openjdk.test.nio")
 				.addProjectToClasspath("openjdk.test.concurrent")
 				.addJarToClasspath(mauveJar)
-				.addJarToClasspath(fileSystemJar)
-				.setAbortIfOutOfMemory(false)
+				.addJarToClasspath(fileSystemJar);
+				
+		if (isTimeBasedLoadTest) { 
+			loadTestInvocation = loadTestInvocation.setTimeLimit(timelimitInMins);	// If it's a time based test, stop execution after given time duration
+		}
+		loadTestInvocation = loadTestInvocation.setAbortIfOutOfMemory(false)
 				.addSuite("mix")
-				.setSuiteThreadCount(cpuCount - 2, 10)	
-				.setSuiteNumTests(totalTests * 200) // Leave 1 cpu for the JIT. i for GC and set min 20			
-				.setSuiteInventory(inventoryFile) 			// Point at the file which lists the tests
-				.setSuiteRandomSelection();		  			// Randomly pick the next test each time
+				.setSuiteThreadCount(cpuCount - 2, 10); 
+		if (isTimeBasedLoadTest) { 
+			loadTestInvocation = loadTestInvocation
+				.setSuiteNumTests(totalTests * 80000000 * testCountMultiplier); 	// If it's a time based test, run a very large load
+		} else {  
+			loadTestInvocation = loadTestInvocation
+				.setSuiteNumTests(totalTests * 200);	// Leave 1 cpu for the JIT. i for GC and set min 20
+		}
+		
+		loadTestInvocation = loadTestInvocation
+				.setSuiteInventory(inventoryFile) 	// Point at the file which lists the tests
+				.setSuiteRandomSelection(); 		// Randomly pick the next test each time
```

Comment:
The only values different for a time limited test are the time limit and the number of tests.  It would be easier to follow if only `setTimeLimit` and `setSuiteNumTests` were outside the initlal `loadTestInvocation` definition.

Reply:
I had tried that, but there was a runtime error from STF:  

```
GEN stderr Exception in thread "main" net.adoptopenjdk.stf.StfException: Java invocation built out of sequence. LOAD_TEST_ARGS cannot be set after SUITE
```

That's why the sequence was kept as it is.